From 1a83cd0f80cf2f4e3a552f66088d9def9245d102 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@apache.org>
Date: Fri, 15 Mar 2013 20:12:29 +0000
Subject: [PATCH 1299/1357] MR1: MAPREDUCE-5073. TestJobStatusPersistency.testPersistency fails on JDK7. (sandyr via tucu)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1457093 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit d5ba7c6a40b66f4cbd51445cc89e0fc79d16cf01)

Author: Sandy Ryza
Reason: Fix failing jdk7 tests
Ref: CDH-10973
---
 .../hadoop/mapred/TestJobStatusPersistency.java    |   19 +++++++++----------
 1 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapred/TestJobStatusPersistency.java b/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapred/TestJobStatusPersistency.java
index d454a46..5b0abd0 100644
--- a/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapred/TestJobStatusPersistency.java
+++ b/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapred/TestJobStatusPersistency.java
@@ -38,6 +38,13 @@ public class TestJobStatusPersistency extends ClusterMapReduceTestCase {
   protected void setUp() throws Exception {
     // Don't start anything by default
   }
+  
+  @Override
+  protected void tearDown() throws Exception {
+    super.tearDown();
+    FileSystem fs = FileSystem.getLocal(new JobConf());
+    fs.delete(TEST_DIR, true);
+  }
 
   private JobID runJob() throws Exception {
     OutputStream os = getFileSystem().create(new Path(getInputDir(), "text.txt"));
@@ -87,7 +94,7 @@ public class TestJobStatusPersistency extends ClusterMapReduceTestCase {
     Properties config = new Properties();
     config.setProperty("mapred.job.tracker.persist.jobstatus.active", "true");
     config.setProperty("mapred.job.tracker.persist.jobstatus.hours", "1");
-    startCluster(false, config);
+    startCluster(true, config);
     JobID jobId = runJob();
     JobClient jc = new JobClient(createJobConf());
     RunningJob rj0 = jc.getJob(jobId);
@@ -118,11 +125,8 @@ public class TestJobStatusPersistency extends ClusterMapReduceTestCase {
   /**
    * Test if the completed job status is persisted to localfs.
    */
-  public void testLocalPersistency() throws Exception {
+  public void testLocalPersistency() throws Exception {    
     FileSystem fs = FileSystem.getLocal(new JobConf());
-    
-    fs.delete(TEST_DIR, true);
-    
     Properties config = new Properties();
     config.setProperty("mapred.job.tracker.persist.jobstatus.active", "true");
     config.setProperty("mapred.job.tracker.persist.jobstatus.hours", "1");
@@ -137,7 +141,6 @@ public class TestJobStatusPersistency extends ClusterMapReduceTestCase {
     // check if the local fs has the data
     Path jobInfo = new Path(TEST_DIR, rj.getID() + ".info");
     assertTrue("Missing job info from the local fs", fs.exists(jobInfo));
-    fs.delete(TEST_DIR, true);
   }
 
   /**
@@ -152,10 +155,6 @@ public class TestJobStatusPersistency extends ClusterMapReduceTestCase {
     try {
       FileSystem fs = FileSystem.getLocal(new JobConf());
 
-      if (fs.exists(TEST_DIR) && !fs.delete(TEST_DIR, true)) {
-        fail("Cannot delete TEST_DIR!");
-      }
-
       if (fs.mkdirs(new Path(TEST_DIR, parent))) {
         if (!(new File(parent.toUri().getPath()).setWritable(false, false))) {
           fail("Cannot chmod parent!");
-- 
1.7.0.4

